app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -995,6 +994,25 @@ public void deliverLoadError(Exception e) {
         displayCaseListLoadException(e);
     }
 
+    @Override
+    public void deliverProgress(Integer[] values) {
+        EntityLoadingProgressListener.EntityLoadingProgressPhase phase =
+                EntityLoadingProgressListener.EntityLoadingProgressPhase.fromInt(values[0]);
+        // throttle to not update text too frequently
+        if (values[1] % 100 == 0) {
+            switch (phase) {
+                case PHASE_PROCESSING -> setProgressText(
+                        StringUtils.getStringRobust(this, R.string.entity_list_processing,

```

Comment:
Raw count text inside the view might be a tricky way to view this progress, btw, compared to another form of loading indicator. 

In the big views we've used with cache and index historically, sometimes the length of the view is tens-of-thousands of cases, which is going to make a fairly large and pretty messy looking view. Even just using a % might work better.

Reply:
https://github.com/dimagi/commcare-android/pull/2928/commits/15cf20ad67bc7290a2ef6502acf2d70049244def